deepchem/models/torch_models/neuralode.py

```diff
@@ -0,0 +1,218 @@
+import torch
+import torch.nn as nn
+
+import numpy as np
+import torchdyn
+from torchdyn.core import NeuralODE
+from deepchem.models import TorchModel
+from deepchem.models.losses import L2Loss
+import tqdm
+
+import torch.nn.functional as F
+from deepchem.models.layers import LinearModule
+from deepchem.utils.typing import ActivationFn, OneOrMany
+from typing import Iterable, Optional, Callable, Union, Generator
+
+
+class NeuralODEModel(TorchModel):
+  """A General Neural ODE API to learn dynamics of the system. The Interface
+  allows users to
+  
+  - Configurable the neural net modelling system dynamics in terms of number of layers,
+  dropouts, weight initialisation
+  - Configurable ode solver, allowing to set various sensitivity, adjoint methods for
+  ode solvers
+  - Custom Wrapper NN to NeuralODE enabling to use for deepchem tasks, and fit on deepchem 
+  datasets
+  
+  TODO : Add Usage Example
```

Comment:
Once we have unit tests and usage examples, we can see whether the API is clean and works